RevenueCatUI/Templates/Components/PresentedPartials.swift

```diff
@@ -72,12 +72,23 @@ extension PresentedPartial {
     static func buildPartial(
         state: ComponentViewState,
         condition: ScreenCondition,
+        isEligibleForIntroOffer: Bool,
         with presentedOverrides: PresentedOverrides<Self>?
     ) -> Self? {
-        let conditionPartial = buildConditionPartial(for: condition, with: presentedOverrides)
+        var conditionPartial = buildConditionPartial(for: condition, with: presentedOverrides)
 
-        guard case .selected = state else { return conditionPartial }
-        return Self.combine(conditionPartial, with: presentedOverrides?.states?.selected)
+        if isEligibleForIntroOffer {
+            conditionPartial = Self.combine(conditionPartial, with: presentedOverrides?.introOffer)
+        }
```

Comment:
This is the core part of this change.

The `introOffer` partial will be applied **after** the screen class and **before** the selected state if an intro offer is available.

Any of the components that use partials (text, image, and stack) will gain this ability.